src/include/types.h

```diff
@@ -493,11 +495,12 @@ struct errorcode32_t {
   }
 
   void encode(bufferlist &bl) const {
-    ::encode(code, bl);
+    __s32 newcode = hostos_to_ceph_errno(code); 
+    ::encode(newcode, bl);
```

Comment:
This was teh weird missing piece in the earlier version. Do you mind separating it into a separate patch?  Eveything else is fixing the build and cmake-related regresssions.  This is a change in behavior so that the the server-side can also use non-Linux errno values.